source/fundamentals/data-formats/documents.txt

```diff
@@ -14,34 +14,351 @@ Documents
+:doc:`CRUD guide </fundamentals/crud>`.
+
+For more information on this class, see the
+:java-docs:`BsonDocument API documentation </apidocs/bson/org/bson/BsonDocument.html>`.
 
 BasicDBObject
 -------------
 
-- Description
-- Code example of declaring one and appending fields
-- Link to Fundamentals > CRUD for more information
-- Link to API documentation
+The ``BasicDBObject`` class allows you to access and manipulate document data
+using Java types. We recommend that you avoid using this class unless you
+are migrating an application from that uses an older driver version. In
+addition to deprecation, this class has the following limitations:
+
+- This class implements the ``DBObject`` interface, so you cannot extend its
+  API without breaking binary compatibility.
+- ``BasicDBObject`` does not implement ``Map<String, Object>`` and therefore
+  lacks the associated data access and manipulation convenience methods.
+
+In the following code snippet, we show how to instantiate and build a sample
+``BasicDBObject`` instance that contains a variety of data types:
+
+.. code-block:: java
+   :copyable: true
+
+   BasicDBObject author = new BasicDBObject("_id", new ObjectId())
+       .append("name", "Gabriel García Márquez")
+       .append("lastUpdated", new Date(120, 2, 26))
+       .append("novels", Arrays.asList(
+           new BasicDBObject("title", "One Hundred Years of Solitude").append("yearPublished", 1967),
+           new BasicDBObject("title", "Chronicle of a Death Foretold").append("yearPublished", 1981),
+           new BasicDBObject("title", "Love in the Time of Cholera").append("yearPublished", 1985)));
+
+You can insert this document into a collection using the following code:
+
+.. code-block:: java
+   :copyable: true
+
+   // MongoClient mongoClient = ...
+
+   MongoDatabase database = mongoClient.getDatabase("fundamentals_data");
+   MongoCollection<BasicDBObject> collection = database.getCollection("authors", BasicDBObject.class);
+
+   InsertOneResult result = collection.insertOne(author);
+
+Once you perform a successful insert, you can retrieve the sample document
+data from the collection using the following code:
+
+.. code-block:: java
+   :copyable: true
+
+   import com.mongodb.client.model.Filters;
+   ...
+
+   BasicDBObject doc = collection.find(Filters.eq("novels.yearPublished", 1967)).first();
+   if (doc != null) {
+       System.out.println("_id: " + doc.getObjectId("_id")
+           + ", name: " + doc.getString("name")
+           + ", lastUpdated: " + doc.getDate("lastUpdated"));
+
+       BasicDBList novels = (BasicDBList) doc.get("novels");
+       if (novels != null) {
+           BasicDBObject[] novelArr = novels.toArray(new BasicDBObject[0]);
+           for (BasicDBObject novel : novelArr) {
+               System.out.println("title: " + novel.getString("title")
+               + ", yearPublished: " + novel.getInt("yearPublished"));
+           }
+       }
+   }
```

Comment:
Similar thoughts as I had for the Document class -- should we explain the usage and constraints of the BasicDBObject type getters?

Reply:
Addressed in a note directive